iothub/device/src/Transport/AmqpIoT/AmqpIoTErrorAdapter.cs

```diff
@@ -77,7 +77,7 @@ public static Exception ToIotHubClientContract(AmqpException amqpException)
             }
             else if (Equals(AmqpErrorCode.DecodeError, amqpSymbol))
             {
-                return new AmqpIoTResourceException(message, amqpException);
+                return new IotHubException(message, amqpException);
```

Comment:
`DecodeError` is not something that can be fixed on retry; should be mapped to `IotHubException`